### ISSUE TITLE
Can we put a queue in the constructor and if we change the queue, will the queue itself changes ?

### ISSUE THREAD

**Issue by Cattttw** [nus-cs2030/2021-s2#366]
# Summary
Put a summary of the question

## Description 
See as attached. I have a queue named "served" in the main function and after I put this queue in the constructor and have a Serve class, will the queue "served" in Main change?

## Screenshots (if any): 
Insert Images here if necessary
[
<img width="1143" alt="Screenshot 2021-04-26 at 8 18 01 PM" src="https://user-images.githubusercontent.com/76734612/116081201-8fbd0200-a6cc-11eb-933e-8da8a79b99ce.png">
](url)

**Comment by ARPspoofing** [comment]
Yes it would change if you modify the same queue.

**Comment by High325** [comment]
Yes it would modify the List. While the List would have the private final modifiers when you declare it as an attribute in your Event class. what would be immutable would be the pointer to the List, not the contents of the List. The pointer would point to a List class that is mutable.

**Comment by ctako** [comment]
Yes it would

**Comment by gato0014** [comment]
Yes, it's similar to list and array.

**Comment by MatthiasTanWeiSheng** [comment]
yes it would

**Comment by amaslua** [comment]
yes u can, as it's similar to an array, so you are changing the contents but not the pointer
